index.bs

```diff
@@ -11113,48 +11108,85 @@ is the concatenation of the [=interface=]’s
-        then the property exists on it.
-*   The value of the property is that which is obtained by [=converted to an ECMAScript value|converting=] the [=constant=]’s IDL value to an ECMAScript value.
-*   The property has attributes { \[[Writable]]: <emu-val>false</emu-val>, \[[Enumerable]]: <emu-val>true</emu-val>, \[[Configurable]]: <emu-val>false</emu-val> }.
+<div algorithm>
+    To <dfn>define the constants</dfn> of [=interface=] or [=namespace=] |definition| on |target|,
+    given [=Realm=] |realm|, run the following steps:
 
-In addition, a property with the same characteristics must
-exist on the [=interface object=] or the [=legacy callback interface object=],
-if either of those objects exists.
+    1.  [=list/For each=] [=constant=] |const| that is a [=member=] of |definition|:
+        1.  If |const| is not [=exposed=] in |realm|, then [=iteration/continue=].
+        1.  Let |value| be the result of [=converted to an ECMAScript value|converting=]
+            |const|’s IDL value to an ECMAScript value.
+        1.  Let |desc| be the PropertyDescriptor{\[[Writable]]: <emu-val>false</emu-val>,
+            \[[Enumerable]]: <emu-val>true</emu-val>, \[[Configurable]]: <emu-val>false</emu-val>,
+            \[[Value]]: |value|}.
+        1.  Let |id| be |const|'s [=identifier=].
+        1.  Perform [=!=] <a abstract-op>DefinePropertyOrThrow</a>(|target|, |id|, |desc|).
+</div>
 
 
 <h4 id="es-attributes">Attributes</h4>
 
-For each [=exposed=] [=attribute=] of the [=interface=]
-there must exist a corresponding property.
-The characteristics of this property are as follows:
-
-*   The name of the property is the [=identifier=] of the [=attribute=].
-*   The location of the property is determined as follows:
-    *   If the attribute is a [=static attribute=],
-        then there is a single corresponding property and it exists on the interface’s [=interface object=].
-    *   Otherwise, if the attribute is [=unforgeable=] on
-        the interface or if the interface was declared with the [{{Global}}] [=extended attribute=],
-        then the property exists on every object that implements the interface.
-    *   Otherwise, the property exists solely on the interface’s [=interface prototype object=].
-*   The property has attributes { \[[Get]]: |G|, \[[Set]]: |S|, \[[Enumerable]]: <emu-val>true</emu-val>, \[[Configurable]]: |configurable| },
-    where:
-    *   |configurable| is <emu-val>false</emu-val> if the attribute was
-        declared with the [{{Unforgeable}}] extended attribute and <emu-val>true</emu-val> otherwise;
-    *   |G| is the [=attribute getter=] created given the attribute, the interface,
-        and the [=relevant Realm=] of the object that is the location of the property; and
-    *   |S| is the [=attribute setter=] created given the attribute, the interface,
-        and the [=relevant Realm=] of the object that is the location of the property.
+[=Static attributes=] are exposed on the [=interface object=].
```

Comment:
I think here it's worth calling out that "define the unforgeable regular attributes" doesn't yet have any callers. Perhaps with a normative statement like "The implementation must perform define the unforgeable regular attributes on every instance created of every interface type. NOTE: we hope to turn this declarative into imperative later." (My phrasing is bad; end of long day; please phrase better :))

Reply:
I guess also another "must" for what to do with globals.

Reply:
Fixed in 6b7ecd0b1ed658ba5260a14ad9b7c4588aa24d0a.